### PR TITLE
add real versions to controller main.go

### PR DESCRIPTION
**Pull request by jaypipes**
I'd left a TODO in the main.go template for controllers to fill in their
GitVersion, GitCommit and BuildDate information properly. Just now
getting around to this...

Issue: aws-controllers-k8s/community#1171

Signed-off-by: Jay Pipes <jaypipes@gmail.com>

By submitting this pull request, I confirm that my contribution is made
under the terms of the Apache 2.0 license.

### CODE REVIEW

File: templates/cmd/controller/main.go.tpl [hunk 1]
```diff
@@ -102,7 +103,11 @@ func main() {
 	)
 	sc := ackrt.NewServiceController(
 		awsServiceAlias, awsServiceAPIGroup, awsServiceEndpointsID,
-		ackrt.VersionInfo{},	// TODO: populate version info
+		ackrt.VersionInfo{
+			version.GitCommit,
```
Comment (vijtrip2) [inline templates/cmd/controller/main.go.tpl:107]: I see these values as empty in the generated controllers right now. What am i missing?

Comment (RedbackThomson) [reply templates/cmd/controller/main.go.tpl:107]: Yeah the LDFLAGS for individual controllers are not being loaded correctly. [You can see](https://github.com/aws-controllers-k8s/s3-controller/blob/main/Makefile#L7-L12) the version is pinned at `v0.0.0` and it's modifying variables that are not at that location (it's trying to reference them in main). Should probably model these flags after the [LDFLAGS in code-generator](https://github.com/aws-controllers-k8s/code-generator/blob/main/Makefile#L9-L15). These one load the version from the dirty git tag (which is clean when built through CI/CD), and properly use import paths for the variable replacement.

Comment (RedbackThomson) [reply templates/cmd/controller/main.go.tpl:107]: The Makefile is not included in `code-generator`, but IMO it should be. There isn't anything in there that is service-specific anyway, so I don't see any problem with doing so.

Comment (vijtrip2) [reply templates/cmd/controller/main.go.tpl:107]: >The Makefile is not included in code-generator, but IMO it should be.

For the most part, yes! But a service team may modify their Makefile and we would not wanna override it from code-generator in that case. Solution for that problem can be custom hooks for adding new targets to auto-generated Makefile.

Comment (vijtrip2) [reply templates/cmd/controller/main.go.tpl:107]: Got it now. These variables are set at build-time.
So this PR looks good to be merged however we still need to fix Makefile to pass the correct ldflags.

**Comment by vijtrip2** [comment]
/approve

@RedbackThomson feel free to merge if you have no concerns.

**Comment by RedbackThomson** [comment]
> /approve
> 
> 
> 
> @RedbackThomson feel free to merge if you have no concerns.

I'd prefer not to have to modify every repositories Makefiles. I would rather we have them code generated tbh

**Comment by vijtrip2** [comment]
> I'd prefer not to have to modify every repositories Makefiles. I would rather we have them code generated tbh

I think we won't need to. This PR will correctly use LDFlags from controller/pkg/version in main.go and the [Dockerfile](https://github.com/aws-controllers-k8s/code-generator/blob/main/Dockerfile#L41) correctly sets those LDFLAGS when controller binary is built.

The LDFlags in controller Makefile are not used anywhere because binary is built in DockerFile which does not use Makefile.

**Comment by RedbackThomson** [comment]
Oh woah I did not notice that! That's confusing, but yay. I'm happy, then!
/lgtm

**Pull request merged**